### PR TITLE
Consistent region/subregion/continent variable naming

### PR DESCRIPTION
**Pull request by Kiel-SFU**

### CODE REVIEW

File: scripts/SpecifiedAnnualDemand.py [hunk 1]
```diff
@@ -62,15 +62,15 @@ def getUsaSpecifiedAnnualDemand():
     #holds output data
     outData = []
 
-    top_level_region = functions.openYaml().get('regions')[0]
+    continent = functions.openYaml().get('continent')
```
Comment (trevorb1) [inline scripts/SpecifiedAnnualDemand.py:65]: Move to top of function?

File: scripts/SpecifiedDemandProfile.py [hunk 2]
```diff
@@ -16,8 +16,8 @@ def main():
     seasons = functions.openYaml().get('seasons')
 
     # Parameters to print over
-    region = functions.openYaml().get('regions')[0]
-    subregions = (functions.openYaml().get('subregions_dictionary'))['CAN'] # Canadian subregions
+    continent = functions.openYaml().get('continent')
+    canSubregions = (functions.openYaml().get('subregions_dictionary'))['CAN'] # Canadian subregions
```
Comment (trevorb1) [inline scripts/SpecifiedDemandProfile.py:20]: brackets

File: scripts/SpecifiedDemandProfile.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def getUsaSpecifiedDemandProfile():
     #holds output data
     outData = []
 
-    top_level_region = functions.openYaml().get('regions')[0]
+    continent = functions.openYaml().get('continent')
```
Comment (trevorb1) [inline scripts/SpecifiedDemandProfile.py:93]: Move to top of function?

File: scripts/StorageCosts.py [hunk 4]
```diff
@@ -13,8 +13,8 @@ def main():
     # OUTPUT: none
 
     # Parameters to print over
-    region = functions.openYaml().get('regions')[0]
-    subregions = ((functions.openYaml().get('subregions_dictionary'))['CAN']).keys() # Canadian subregions
+    continent = functions.openYaml().get('continent')
+    canSubregions = ((functions.openYaml().get('subregions_dictionary'))['CAN']).keys() # Canadian subregions
```
Comment (trevorb1) [inline scripts/StorageCosts.py:17]: brackets

File: scripts/config.yaml [hunk 5]
```diff
@@ -5,9 +5,8 @@ seasons:
...
   'F': [ 9, 10, 11 ]
 
-# only one region in this model
-regions: 
-  - 'NAmerica'
+# only one continent ('r' in osemosys, for 'region') in this model
+continent: 'NAmerica'
 
 # Provinces and states to subregions mapping
 subregions_dictionary:
```
Comment (trevorb1) [inline scripts/config.yaml:12]: Shouldn't this be `regions_dictionary`? Also, if you wanted to shorten the name to just `regions` or `region_dict` or something like that, feel free to! Up to you though, just a suggestion if writing `subregions_dictionary` was getting a little tedious to write :)

**Review by trevorb1 — CHANGES_REQUESTED** [review]
Looking good @Kiel-SFU ! A couple of notes down below... 

1. You were adding in an extra set of brackets when reading in the continent name from the yaml file. I think I pointed out most of them, but please do a double check. 
2. Can we move all the `functions.openYaml().get(...)` lines to the top of the function they are used in please? In some places we are writing theses statements in two different places within one function, so it can be hard to follow. 
3. When we comment what the final printed dataframe columns will be (see pic below), we originally wrote `region`. As you pointed out in our call earlier today, this can be confusing since we need to keep the name `REGION` as otoole needs it, but for the rest of the code we refer to region as a country (`CAN` or `USA`). You suggested to change this comment to read `continent` instead of region (which is totally fine), but it looks like you only did this for about half of the instances? Not sure if this was a mistake or not? (For example see CapacityToActivityUnit, InOutActivityRatio, RETags, ReserveMargin, ResidualCapacity, SpecifiedDemandProfile ...). Since we are gonna have to do a thorough review to insure we are consistent with this comment, lets go back to calling it `region`. There is no real right answer here, other then `region` in this context is clearer to me. :) 

image

4. Can you please add a note in the README.md file that quickly highlights our naming convention for what we consider a region, subregion, and province? Basically what is in [this comment](https://github.com/DeltaE/Canada-U.S.-ElecTrade/issues/81#issuecomment-1014995638). Doesnt have to be fancy or anything, more so just as a quick reference so we dont lose the information

File: scripts/README.md [hunk 6]
```diff
@@ -2,3 +2,11 @@
 
 ## Contents
 Included in this folder are the scripts used to pre-process, and a folder for post-processing scripts. Info on each script can be found in the header files, and the src/snakemake file.
+
+### Note on naming convention
+In these scripts, these words are used to indicated different regional scales:
+
+- Continent: The smallest scale. In this model there is only one; NAmerica. Note that this level is called REGION by OSeMOSYS, and so CSV files are populated with this in mind
```
Comment (trevorb1) [inline scripts/README.md:9]: Not sure what `The smallest scale` means?

File: scripts/README.md [hunk 7]
```diff
@@ -2,3 +2,11 @@
...
+- Region: Currently equivalent to 'country', this model has a US region and a Canada region
+- Subregion: This model is broken up into many subregions (such as Mountain West, Mid-Atlantic, Midwest...) each of which contain one or more provinces
```
Comment (trevorb1) [inline scripts/README.md:11]: Instead of saying `This model`, lets say `Each Region is broken into ...`

File: scripts/MakeSets.py [hunk 8]
```diff
@@ -11,14 +11,14 @@ def main():
     ## MODEL PARAMETERS
     ####################################
 
-    continent = [functions.openYaml().get('continent')]
-    emissions = functions.openYaml().get('emissions')
-    techsMaster = functions.openYaml().get('techs_master')
-    rnwFuels = functions.openYaml().get('rnw_fuels')
-    mineFuels = functions.openYaml().get('mine_fuels')
-    stoTechs = functions.openYaml().get('sto_techs')
+    continentList = [functions.getFromYaml('continent')]
```
Comment (trevorb1) [inline scripts/MakeSets.py:14]: brackets

**Review by trevorb1 — CHANGES_REQUESTED** [review]
Almost there ! Most of the code changes look good to me ! Please make sure you rerun the workflow before every pull request though. There is currently an error preventing it from running

`(osemosys) trevorb1@DESKTOP-HCG4NHT:~/repositories/Canada-U.S.-ElecTrade/src$ snakemake dataFile -c1 -F`

`KeyError in line 85 of /home/trevorb1/repositories/Canada-U.S.-ElecTrade/src/snakefile:
'subregions_dictionary' File "/home/trevorb1/repositories/Canada-U.S.-ElecTrade/src/snakefile", line 85, in <module>`

From the snakefile
image

File: scripts/MakeSets.py (hunk 8 above)
Comment (Kiel-SFU) [reply scripts/MakeSets.py]: I think brackets to turn this into a list are necessary either here, or where continentList is called
